src/validation/userValidation.js

```diff
@@ -0,0 +1,153 @@
+      password: new passwordComplexity(complexityOptions).required()
+        .label('Password is required. \n It should be more than 8 characters, and should include at least a capital letter, and a number'),
+      firstName: joi.string().min(3).max(25).required()
+        .label('Please enter a valid firstname \n the field must not be empty and it must be more than 2 letters'),
+      lastName: joi.string().min(3).max(25).required()
+        .label('Please enter a valid lastname \n the field must not be empty and it must be more than 2 letters'),
+      companyName: joi.string().min(3).max(40).required()
+        .label('Please add your company name'),
+      companySizeId: joi.number().integer().positive().required()
+        .label('please select a company size'),
+      companyPlanId: joi.number().integer().positive()
+        .required()
+        .label('please select a plan (1:silver, 2:gold or 3:platinum'),
+      companyAddress: joi.string().min(10).max(60).regex(/^[\w',-\\/.\s]*$/)
+        .required()
+        .label('Please enter a valid address that is within 10 to 60 letters long'),
+    };
+    // Once user inputs are validated, move into server
+    const { error } = joi.validate({ ...userObject }, schema);
+    if (error) {
+      throw error;
+    }
+    return true;
+  }
+
+  /**
+     * Validates user paramenters upon registration
+     *
+     * @param {object} userObject - The user object
+     * @param {object} res - The user response object
+     * @returns {object} - returns an object (error or response).
+     */
+  static async userLogin(userObject) {
+    // joi parameters to test against user inputs
+    const schema = {
+      email: joi.string().email().required()
+        .label('Please enter a valid company email address'),
+      password: new passwordComplexity(complexityOptions).required()
+        .label('Password is required. \n It should be more than 8 characters, and should include at least a capital letter, and a number'),
+    };
+    // Once user inputs are validated, move into server
+    const { error } = joi.validate({ ...userObject }, schema);
+    if (error) {
+      throw error;
+    }
+    return true;
+  }
+
+  /**
+   *  Dummy callback function for validation tests
+   * Use this function as a placeholder for controllers
+   * during testing of validations whenever the controller
+   * being validated for is not yet implemented
+   *
+   * e.g: @ route:-
+   * userRouter.post('/auth/signup', userValidation.signup, userValidation.dummy);
+   * @param {object} req request from endpoint
+   * @param {object} res - response of method
+   * @return {object} - returns an object
+   */
+  static dummy(req, res) {
+    try {
+      // outdated response values, dummy parameter used for testing only
+      successResponse(res, 'Success', 200);
+    } catch (error) {
+      const status = error.status || 500;
+      errorResponse(res, { code: status, message: error.message });
+    }
+  }
+}
```

Comment:
Thanks for your observations @King-Success . It would seem like these files were added and re-modified while the team worked on their respective feature as I didn't create this file, I also didn't work on validations for this feature, however, I'd look into this and give you feedback

Reply:
Ok great @daylay92

Reply:
Hi @King-Success .Its been sorted.

Reply:
Yes I can see that, Well done Ayo